### ISSUE TITLE
Support for noarch conda packages?

### ISSUE THREAD

**Issue by ceball** [conda/constructor#86]
https://docs.continuum.io/anaconda-repository/admin/noarch-packages includes this:

> NOTE: Noarch packages are not compatible with Anaconda constructor. If you intend to use the packages with Anaconda constructor, build the packages for specific operating systems.

I couldn't find an issue about this here, so I wanted to check: is this still the case?

**Comment by mingwandroid** [comment]
If it is we should fix it.

**Comment by ceball** [comment]
Sorry, I missed this near the end of the README:

> Constructor does not work with noarch-Python packages. All conda packages must be available for the platform you are building the installer for.

In that case my question is: can this be a feature request, or is it not part of the plan?

Thanks

**Comment by ericdill** [comment]
@mingwandroid @csoja is noarch support going to be added to constructor?

**Comment by parente** [comment]
To add a concrete example: If I add a norarch package to my constructor definition today, it builds fine but upon running it, all the noarch packages wind up in `<env root>/site-packages` instead of in `<env root>/lib/python3.6/site-packages` (or whatever version of Python I've specified in the constructor). As a result, they're not in the python path and not importable. In contrast, when I `conda install` a noarch package, it winds up in `<env root>/lib/python<version>/site-packages`.

Perhaps a path prefix missing somewhere in the noarch case?

**Comment by mingwandroid** [comment]
I've been cautioning people not to use `noarch: python` for a while now. It's just not ready for prime-time I'm afraid and this is a good example of that (others include permissions with .pyc files, menu shortcuts .. the list probably goes on and on).

It'd be nice to get some of this fixed, contributions are welcome.

**Comment by parente** [comment]
Interesting. We're just starting to dabble in the noarch space and I, at least, was not aware there was a litany of issue.

Happy to try to contribute a fix for this initial problem here, but no one should wait on me if they get to it first.

**Comment by ericdill** [comment]
@mingwandroid are you recommending against the use of `noarch: python` in general or just in the context of constructor?

**Comment by mingwandroid** [comment]
I wouldn't use it until all of the issues are fixed. It's a bit chicken and egg though, I mean if no one uses it then no one will run into the problems and attempt to fix them. I'm not even sure what can be done about .pyc files in locked down scenarios.

Everyone's use-case is different though. With my Anaconda Distro hat on I have to take a very conservative approach.

**Comment by mbargull** [comment]
@mingwandroid I've a strong interest to push `noarch: python` support generally. Would you mind if we open up some meta "`noarch: python` issues"-issue over at https://github.com/conda/conda/issues where we compose a list of problems you/we see with the current `noarch: python`-support. We could then tackle each problem in separate issues/**tests**/PRs -- I'm quite willing to contribute to this!
(For menu shortcuts we have https://github.com/conda/conda/issues/5153 which could be solved if https://github.com/ContinuumIO/menuinst/issues/48 gets traction.)

**Comment by danielfrg** [comment]
This is actually an issue now that conda-forge is starting to push some pkgs with `noarch: python`. 

Right now I think its impossible to create a working constructor based on new pkgs from conda-forge, for example [six wont work](https://github.com/conda-forge/six-feedstock/pull/8/files) now that it has `noarch: python`. Is there a way to tell constructor to completely ignore noarch pkgs?

**Comment by mingwandroid** [comment]
@danielfrg, not to my knowledge. As a team we're all a bit too busy to fix this. The fix would seem to necessarily intrude heavily into the solver code which would be probably be unpleasant for all involved.

@mbargull, sure, sounds good and thanks for helping out (as usual). Same thing though, we're trying to get the next release out so are all too busy to do anything for this at present. I'll have some time to think about it in a couple of weeks, but I'd love to get your take on how to approach it. I am leaning towards taking the core installer part of conda out of conda (and Python actually) and adding low-level support that would be useful for menuinst to that. This could include conversion from e.g. .png (or .svg?) to platform specific icons.

**Comment by danielfrg** [comment]
I just tested this on `constructor=1.3` that doesn't support noarch pkgs and it worked fine by just pulling from `linux-64` for example.

I am not sure why was this introduced in https://github.com/conda/constructor/pull/43 if it actually doesn't work, i guess it work for noarch pkgs that are not python libs but most of them are noarch-python pkgs so it break most of the time. And it actually doesn't break, like other people reported the installer finishes but libs are impossible to import.

I guess the solution would be for a flag to remove the noarch path of the channel (https://github.com/conda/constructor/blob/master/constructor/fcp.py#L181)

Even better would be to actually support noarch-python pkgs but I am afraid i don't understand enough to know what the problem is. Is it the postlink script of the noarch-python pkgs?

**Comment by mingwandroid** [comment]
We've moved back to using `conda` in constructor now instead of `libconda` (unfortunately we're mostly working out of a different branch for reasons I will not bore you with: https://github.com/conda/constructor/tree/cross_conda_interface), but yes, since they're in different subdirs, it's probably quite easy to stop `conda` from picking them up, cc'ing @kalefranz?

The work for un-noarch-ifying noarch python packages is done in `conda` at present.

**Comment by mingwandroid** [comment]
The core problem with `constructor` is that it has 3 backends (macOS & linux shell script, macOS pkg and Windows NSIS) and they do not share very much in common. This is why I want to extract the core package installation logic into a (fully static) standalone executable (that is bundled with the installers), then each of the 3 installer front-ends do little more than run this executable.

**Comment by danielfrg** [comment]
Got it. Thanks for explaining that, i know there is a lot of context i dont have. Hopefully there is an easy fix for having constructor working again with conda-forge now that they have some noarch pkgs. other than manually listing everysingle version on the constructor, that is really not that bad.

**Comment by ericdill** [comment]
@mingwandroid One short-term fix is just to blacklist all `noarch: python` packages from constructor. This is a pretty simple fix as you can see in #115. I think this would be a nice short-term solution so that we can still continue to use constructor like we have been until constructor fully supports the noarch python packages. Thoughts?

**Comment by mingwandroid** [comment]
I'm fine with it, we may need @ilanschnell to review and merge it though (presumably you'd need a release/tag with this change?), unless can target our new branch where I'm happy to review and merge it.

**Comment by ericdill** [comment]
> presumably you'd need a release/tag with this change?

Selfishly, this patch means that I've already figured out what *I* need to fix and build internally so that I can continue using constructor at my day-job.

Putting my open-source contributor hat on, it'd be a shame to not release/tag with this update since, without it, anyone that is currently using constructor 1.5.5 *and* targeting the conda-forge channel will start producing "bad" constructor installer scripts if they unknowingly pull in a noarch python package. I'd like to suggest branching at 1.5.5, applying this patch and releasing a 1.5.6.

> unless can target our new branch where I'm happy to review and merge it.

What's the new branch? Are you going to be cutting releases from that new branch?

What's the deal with the 1.6.0 tag? I don't see that version available anywhere?

**Comment by mingwandroid** [comment]
Here is the branch: https://github.com/conda/constructor/tree/cross_conda_interface

I don't know the answers to the rest of your questions, I'll try to bring them up with people who might.

**Comment by ericdill** [comment]
@mingwandroid any updates here? Would be lovely to get #115 in and a new version made for the folks starting to use noarch packages

**Comment by mingwandroid** [comment]
I'm on vacation, pinging @csoja.

**Comment by ericdill** [comment]
Oh have fun 😎  ⛰ 🏖 🍸 ! 

Sorry for interrupting your vacation 😞

**Comment by mingwandroid** [comment]
No problem! Hopefully we can get constructor releases sorted out for you soon.

**Comment by kalefranz** [comment]
@mingwandroid 

> I've been cautioning people not to use `noarch: python` for a while now. It's just not ready for prime-time I'm afraid

Are there any bugs with conda proper that you're aware of, or is that statement more specific to constructor?

**Comment by kalefranz** [comment]
@mbargull Same question for you.  Are you aware of any implementation problems or bugs for conda specifically w.r.t. `python: noarch`?

The *only* issue/bug that I'm currently aware of relates to these packages already installed into and environment, and the switching the python version of an environment.

**Comment by mingwandroid** [comment]
Other than constructor (which is hugely important both for AD and anyone wanting to make installers) there are issues reported/suspected/rumoured with `noarch: python` in:

1. menuinst.
2. Use as dependencies.
3. Multiuser/permissions.
4. Pinned, old conda versions.

Hopefully we can get these all investigated and addressed if appropriate.

Sorry this is a little vague.

**Comment by kalefranz** [comment]
Just want to avoid spreading FUD about conda.  Issues no. 2 and no. 3 could potentially be something going on with conda.  Issues no. 1 and no. 4 are out of the conda repository's control.

Are there any tickets I missed regarding issues no. 2 and no. 3 with conda?  Again, just trying to avoid FUD, which the vagueness can engender.

**Comment by mingwandroid** [comment]
My concern is not about just conda though, so it's not 'FUD about conda', it's a balid concern for the whole of the AD ecosystem.

The issues may well turn out to be in other bits of software, that's fine.

From my POV, to be able to say that `noarch: python` works fine for AD I need to be able to build a Miniconda based around as many noarch packages as possible and for that Minconda to work correctly in all the situations we require it to. I hope you understand this is *not* in any way an attack on `conda`.

**Comment by ericdill** [comment]
What is this FUD?

**Comment by csoja** [comment]
Fear, uncertainty and doubt (often shortened to FUD)  (wikipedia has a nice page explaining it in more detail)

**Comment by kalefranz** [comment]
Interesting, I'd never actually read the wikipedia entry before.  Didn't realize there was a maliciousness to the FUD connotation.  That's not at all what I was implying either.

@mingwandroid For more context here, someone today brought up noarch package in a meeting, and how conda may not have all the quirks worked out yet, pointing to the statement of yours I quoted above as evidence.  There are definitely a lot of moving parts in the AD ecosystem as a whole, and I guess we just need to be a bit more nuanced here.  If there are bugs in conda 4.3 proper regarding noarch (besides the python version change one), I just want to make sure I know about them so I can jump on them immediately.

For the community more broadly, using `noarch: python` packages in constructor has a lot of implications, including making conda a hard-dependency of every constructor installer, just like the python interpreter is included in every constructor installer (even what you're building the installer for doesn't necessarily need python).  Moving farther down this path is or at least should be an option.  There's also various discussion about moving parts of conda to something that doesn't require a python interpreter so that we can compile a single-file binary executable to be used in situations like this.

**Comment by ericdill** [comment]
Hi @kalefranz @mingwandroid @csoja @danielfrg --

Are there any updates for constructor + ``noarch: python``? Right now constructor is currently in a broken state with respect to this feature of conda/conda-build. Binaries created with constructor will silently be broken if the user intentionally or unintentionally pulls in a noarch: python package. It would seem prudent to at least *not* pull from the noarch channel until this major issue is resolved (#115).

@kalefranz -- It seems that Anaconda is not making much use of the ``noarch: python`` feature yet if [the package list of the defaults channel](https://repo.continuum.io/pkgs/free/noarch/) is to be believed. IIUC you are suggesting that ``noarch: python`` is ready for prime time. Is there a different channel that Anaconda is using to distribute ``noarch: python`` packages?

**Comment by jakirkham** [comment]
Was thinking about this a few days ago. Would it be possible to use `conda` or `conda-build` to convert `noarch: python` packages into an architecture/Python version specific package before bundling everything together with `constructor`? If so, that would make `conda` and/or `conda-build` a requirement for creating an installer with `constructor`, but would not require either to actually run the install or use the resulting deployment. Thoughts on this?

**Comment by mingwandroid** [comment]
@ericdill, I should note that none of the packages in AD5 came from the URL you pointed to. This is the correct noarch URL: https://repo.continuum.io/pkgs/main/noarch/

**Comment by ericdill** [comment]
@mingwandroid -- thanks for the corrected link. What's the relationship between ``pro``, ``free`` and ``main``?

@jakirkham -- 👍

**Comment by mingwandroid** [comment]
`pro` is something I've never had much dealings with, I *think* it's mostly deprecated. `free` contains packages pre-dating `AD5`. `main` is the new stuff. Nearly all of the packages are based on forks of `conda-forge` recipes and all C and C++ packages are compiled with the brand new pseudo cross-compilers on macOS and Linux.

**Comment by ericdill** [comment]
@mingwandroid @kalefranz any updates on this?

**Comment by jakirkham** [comment]
Is this still an issue with the latest `constructor`? I actually don't know the answer. Just am curious if people have tried and/or if there is a fix?

**Comment by mingwandroid** [comment]
Nothing to add from me, sorry. I do not like the idea of converting from `noarch` to `arch` though. You end up creating two different versions of the same thing and that seems like bad technical debt.

**Comment by kalefranz** [comment]
From my comment above

> using `noarch: python` packages in constructor has a lot of implications, including making conda a hard-inclusion in every constructor installer, just like the python interpreter is included in every constructor installer (even when what you're building the installer for doesn't necessarily need python).

That still summarizes my current opinion on the issue.

**Comment by CJ-Wright** [comment]
Has there been any progress on this front? `noarch` seems to be [encouraged for conda-forge packaging](https://conda-forge.org/docs/meta.html#building-noarch-packages) so it would be very helpful to have this fixed.

**Comment by mbargull** [comment]
For the long term, I think including `conda` (or a subset of it) would be the most consistent choice, but FYI, for the interim: @jakirkham turned his [comment](https://github.com/conda/constructor/issues/86#issuecomment-335196496) about converting `noarch` packages into an issue over at https://github.com/conda/conda-build/issues/2611.

**Comment by jakirkham** [comment]
While that definitely could turn into something used by `constructor`, there is still value in being able to convert between different types of packages regardless of the intent. One can already do this for the arch packages as long as they are truly platform independent. It's only natural to extend it to the `noarch: generic`  and `noarch: python`.

**Comment by mingwandroid** [comment]
This is something that the community will probably have to address in its entirety of it is to be fixed.

I personally will not have time and AD can and does work around it by not using any `noarch: python` packages.

Even if it were fixed I'd still be very hesitant to make use of it due to permissions issues around .pyc files in multi-user setups.

**Comment by mbargull** [comment]
@mingwandroid: How do those `.pyc` permission issues occur; are they tracked somewhere over at https://github.com/conda/conda/issues?

**Comment by kalefranz** [comment]
@mbargull I think the statement above from @mingwandroid 

> I've been cautioning people not to use noarch: python for a while now. It's just not ready for prime-time I'm afraid and this is a good example of that (others include permissions with .pyc files, menu shortcuts .. the list probably goes on and on).

Was specifically targeted at constructor, and not necessarily conda.  If there are outstanding issues in conda for noarch, I'm not aware of them.

**Comment by cfobel** [comment]
For what it's worth, I've had luck working around the issue on Windows by including the following in a `post-install.bat` script:

```
@echo off
REM Explicitly move noarch packages into `Lib/site-packages` as a workaround to
REM [this issue][i86] with lack of `constructor` support for `noarch` packages.
REM
REM [i86]: https://github.com/conda/constructor/issues/86#issuecomment-330863531
IF EXIST site-packages (
REM Move directory modules.
for /D %%i in (site-packages/*) do IF "%%~xi" == "" (IF NOT EXIST Lib\site-packages\%%i (
    echo Move noarch package: %%i
    move site-packages\%%i Lib\site-packages
))
REM Move file modules.
for %%i in (site-packages/*.py) do IF "%%~xi" == ".py" (IF NOT EXIST Lib\site-packages\%%i (
    echo Move noarch package: %%i
    move site-packages\%%i Lib\site-packages
))
rmdir /S/Q site-packages
)
```

It moves the packages from the `site-packages` directory in the root directory into the proper `Lib/site-packages` location.  Since the packages are in the installed index, conda seems to be happy and the packages can be uninstalled, etc. as normal.

Maybe this hack will be of use to someone else until the common `conda` static executable proposed by @mingwandroid is completed.

**Comment by astrofrog** [comment]
Now that ``noarch: python`` packages are very common, any updates on making constructor work with them?

**Comment by msarahan** [comment]
Some work towards this at https://github.com/conda/constructor/pull/239

This is a short-term goal, but will require significantly more work.

**Comment by jschueller** [comment]
@astrofrog as a workaround noarch packages content need to be copied, see
https://github.com/openturns/otconda/blob/master/post_install.sh

**Comment by ankostis** [comment]
@cfobel & @jschueller: the move-`site-packages` on  `post_install` workaround works, but does not create any entry-points specified in the package (aka executable commands).

Maybe you want to update your suggestions and highlight this significant drawback.

**Comment by ProgramFan** [comment]
I patched the install.py and introduce a preliminary support of noarch:python packages. The idea is to move `site-packages/*` into `lib/pythonX.Y/site-packages`, move `python-scripts/*` into `bin`, and generate shims for entry points. The prefixes are also replaced.

I use them to make packages from conda-forge, pyviz and plotly and it seems to work properly.

**Comment by ProgramFan** [comment]
I am making a PR for this.

**Comment by nehaljwani** [comment]
Alright, time to revive this old thread.

I am working on constructorv3 which will support noarch packages, but not in the way you'd think. This version of constructor will no longer use the .install.py script and will instead use a small, conda executable to perform all the installation operations.

The first Miniconda release with this version of constructor will be made for conda 4.7.x. I have some test installers at https://repo.anaconda.com/pkgs/misc/previews/miniconda/4.7.5/

Dear community members, please test them out. You'll see that the tqdm package in the installers is noarch.

Once the release is successful, the changes will be published to this repository.

**Comment by ProgramFan** [comment]
Nice to see this eventually happens. I am testing it on macos.

**Comment by jschueller** [comment]
@nehaljwani it works fine on Linux x86_64

**Comment by ProgramFan** [comment]
@nehaljwani It works on macOS Mojave: installs fine and tqdm can be imported.

**Comment by ProgramFan** [comment]
I am also curious about whether it is possible to drop python out in the installation if python is not listed as a dependency? @nehaljwani

**Comment by nehaljwani** [comment]
Yes, constructor 3 will remove that forced dependency.

**Comment by astrofrog** [comment]
@nehaljwani - thanks for working on this! I can confirm the Python 3 version for MacOS X works fine and tqdm imports without error. Do you have an estimated time frame for when the changes to constructor will be pushed to this repository? I am interested in using constructor to build an installer for an environment for a SciPy 2019 tutorial (in a little over a week). Is it possible the updated constructor code will be pushed here before then?

**Comment by ProgramFan** [comment]
I am also building python distributions for my institution and we use conda-forge noarch packages. We also distribute our software using constructor and the ability to remove python from the installation tree is highly wanted. Wait to test the new constructor.

BTW, if you want a temporary solution for noarch packages on Linux, you can use my patch in PR #256 by either install it using pip or through https://anaconda.org/programfan/constructor. @astrofrog

**Comment by carterbox** [comment]
@nehaljwani I tested `Miniconda3-4.7.5.rc2-Windows-x86_64.exe` and `Miniconda2-4.7.5.rc2-Windows-x86_64.exe`. I was able to import tqdm and do other conda things (which depend on tqdm). I think that's the last OS you needed testing for?

The anaconda prompt is missing from this Windows installer, so I had to open a cmd window and activate the base conda environment myself with the activate script, but this is a separate problem.

**Comment by nehaljwani** [comment]
The prompt was missing because I forgot to add console_shortcut package to the installer. My bad. 

Thank you for testing! I'm pretty close to releasing the changes, probably by the end of this week.

**Comment by nehaljwani** [comment]
Hello folks, I have another set of preview installers available. Please test these out too, it will be a huge help in making the first release of constructor v3 as quick as possible.

https://repo.anaconda.com/pkgs/misc/previews/anaconda/2019.06/

**Comment by jschueller** [comment]
Works well too on Linux/x86_64

**Comment by ProgramFan** [comment]
Works well on macOS Mojave. Only the .sh installer is tested.

**Comment by jakirkham** [comment]
@carterbox, are you able to retest with [the new installers]( https://github.com/conda/constructor/issues/86#issuecomment-508619131 )?

**Comment by carterbox** [comment]
I was able to install miniconda, import tqdm, and uninstall miniconda with all of Windows installers.

**Comment by jschueller** [comment]
hello @nehaljwani any updates on this ?

**Comment by nehaljwani** [comment]
I have pushed a lot of changes to the v3 branch on this repo.

Please test it out. Also,
- Make sure you have conda >=4.7 in the environment where you install constructor
- You can get the conda.exe files from https://repo.anaconda.com/pkgs/misc/conda-execs/ (content here may change without notice)

**Comment by jschueller** [comment]
@nehaljwani your branch works fine (at least on linux64), I could build/install/test otconda (https://github.com/openturns/otconda) without the post_install workaround.

Here is the recipe patch against constructor-feedstock I used to build constructor:
```
diff --git a/recipe/meta.yaml b/recipe/meta.yaml
index 2275cf6..ea93536 100644
--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,4 +1,4 @@
-{% set version = "2.3.0" %}
+{% set version = "3.0" %}
 {% set sha256 = "81522d7e9b63b66970314e63c1aad69a383541de2d4052fa3e69f892a5bf5dfc" %}
 
 package:
@@ -6,12 +6,11 @@ package:
   version: {{ version }}
 
 source:
-  fn: constructor-{{ version }}.tar.gz
-  url: https://github.com/conda/constructor/archive/{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  git_url: https://github.com/conda/constructor.git 
+  git_rev: v3
 
 build:
-  number: 1000
+  number: 0
   script: python setup.py install --single-version-externally-managed --record=record.txt
   entry_points:
     - constructor = constructor.main:main
@@ -23,7 +22,7 @@ requirements:
 
   run:
     - python
-    - conda >=4.5
+    - conda >=4.7
     - ruamel_yaml
     - pillow >=3.1     # [win]
     - nsis >=3.01      # [win]
@@ -32,7 +31,7 @@ test:
   imports:
     - constructor
   commands:
-    - constructor --test
+    - constructor -h
 
 about:
   home: http://github.com/conda/constructor
```

The only change was adding the required conda exe to the call:
```
 constructor -v --conda-exe /tmp/conda-4.7.5-linux-64.exe .
```

**Comment by jschueller** [comment]
@nehaljwani why arent we allowed to do `constructor --test --conda-exe /tmp/conda-4.7.5-linux-64.exe` without DIRECTORY argument:
```
constructor: error: the following arguments are required: DIRECTORY
```

**Comment by nehaljwani** [comment]
@jschueller Whoops. I might get backlash for saying this, but ... pull requests accepted 😅

**Comment by jschueller** [comment]
@nehaljwani did you fix it already ? I dont see "required=True" for DIRECTORY argument in  constructor/main.py

**Comment by jschueller** [comment]
v3 is now its using argparse instead of the conda options parser, and DIRECTORY is a positional argument so its required, and I dont how if its possible to disable that argument if '--test' is given.

**Comment by ankostis** [comment]
Could it be that with v3 constructor, on Windows at least, `exclude` packages (e.g. `qt`) are reported as missing on installation time, and conda it tries to download them, and chokes because remote operations are not allowed? 
(screenshot would be better if only installation dialog could be resized
![ExcludePackages](https://user-images.githubusercontent.com/501585/61501399-89660600-a9cf-11e9-846c-595f155a84ad.PNG)
...)

Personally can live with it, but better amend the docs to report that the `construct.yaml: exclude` key is not working anymore.

**Comment by FelixS90** [comment]
> I have pushed a lot of changes to the v3 branch on this repo.
> 
> Please test it out. Also,
> 
> * Make sure you have conda >=4.7 in the environment where you install constructor
> * You can get the conda.exe files from https://repo.anaconda.com/pkgs/misc/conda-execs/ (content here may change without notice; since this is still experimental)

Thanks a lot for your efforts!

Although I followed your instructions mentioned above, I receive the following error message during installation,:
![ErrorWindow](https://user-images.githubusercontent.com/9250590/61788936-ac4f4a80-ae13-11e9-8ca9-6115655f9082.png)

Upon clicking on _Ignore_, the installer resumes, however ending up with an incomplete installation (i.e. there is no python.exe whatsoever):
![InstallDir](https://user-images.githubusercontent.com/9250590/61789341-bb82c800-ae14-11e9-9059-8d0a06af7183.png)

From the installation log, it seems that the initial error stems from a **file (_conda-script.py) not being found**.

```
Output folder: C:\Users\m1fsche\test8\Lib
Extract: _nsis.py
Extract: _system_path.py
Output folder: C:\Users\m1fsche\test8\conda-meta
Extract: history
Output folder: C:\Users\m1fsche\test8
Extract: _conda.exe
Output folder: C:\Users\m1fsche\test8\pkgs
Extract: env.txt
Extract: urls
Extract: urls.txt
Extract: post_install.bat
Output folder: C:\Users\m1fsche\test8\pkgs\cache
Extract: 09cdf8bf.json
Extract: 1596f0a5.json
Extract: 346915cb.json
Extract: 36ad50b6.json
Extract: 5afe41e9.json
Extract: d4cdf3a6.json
Extract: f25481b3.json
Extract: faa555db.json
Output folder: C:\Users\m1fsche\test8\pkgs
Output folder: C:\Users\m1fsche\test8\pkgs\_tflow_select-2.3.0-mkl\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\absl-py-0.7.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\albumentations-0.3.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\astor-0.7.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\atomicwrites-1.3.0-py36_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\attrs-19.1.0-py36_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\blas-1.0-mkl\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\blinker-1.4-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\bokeh-1.3.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\ca-certificates-2019.5.15-0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\certifi-2019.6.16-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\clangdev-8.0.0-h1ad3211_2\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\click-7.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\cloudpickle-1.2.1-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\colorama-0.4.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\cycler-0.10.0-py36h009560c_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\cython-0.29.12-py36ha925a31_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\cytoolz-0.10.0-py36he774522_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\dask-0.17.4-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\dask-core-0.17.4-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\decorator-4.4.0-py36_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\dill-0.2.9-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\distributed-1.21.8-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\flask-1.1.1-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\freetype-2.9.1-ha9979f8_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\gast-0.2.2-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\geos-3.7.1-h33f27b4_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\grpcio-1.16.1-py36h351948d_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\h5py-2.9.0-py36h5e291fa_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\hdf5-1.10.4-h7ebc959_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\heapdict-1.0.0-py36_2\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\icc_rt-2019.0.0-h0cc432a_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\icu-58.2-ha66f8fd_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\imageio-2.5.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\imgaug-0.2.9-py_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\importlib_metadata-0.17-py36_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\intel-openmp-2019.4-245\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\itsdangerous-1.1.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\jinja2-2.10.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\joblib-0.13.2-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\jpeg-9b-hb83a4c4_2\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\keras-2.2.4-0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\keras-applications-1.0.8-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\keras-base-2.2.4-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\keras-preprocessing-1.1.0-py_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\kiwisolver-1.1.0-py36ha925a31_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libiconv-1.15-h1df5818_7\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libmklml-2019.0.3-0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libpng-1.6.37-h2a8f88b_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libprotobuf-3.8.0-h7bd577a_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libtiff-4.0.10-hb898794_2\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libxml2-2.9.9-h464c3ec_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\libxslt-1.1.33-h579f668_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\llvmdev-8.0.0-h1acfa63_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\locket-0.2.0-py36hfed976d_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\markdown-3.1.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\markupsafe-1.1.1-py36he774522_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\marshmallow-3.0.0b8-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\matplotlib-3.1.0-py36hc8f65d3_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\mkl-2018.0.3-1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\mkl_fft-1.0.6-py36hdbbee80_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\mkl_random-1.0.1-py36h77b88f5_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\mock-3.0.5-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\more-itertools-7.0.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\msgpack-python-0.6.1-py36h74a9793_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\multiprocess-0.70.8-py36hfa6e2cd_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\networkx-2.3-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\numpy-1.15.4-py36ha559c80_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\numpy-base-1.15.4-py36h8128ebf_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\olefile-0.46-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\opencv-3.3.1-py36h20b85fd_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\openssl-1.1.1c-he774522_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\packaging-19.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pandas-0.24.2-py36ha925a31_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\partd-1.0.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pathos-0.2.4-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\patsy-0.5.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pillow-6.1.0-py36hdc69c19_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pip-19.1.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pluggy-0.12.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pox-0.2.6-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\ppft-1.6.6.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\protobuf-3.8.0-py36h33f27b4_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\psutil-5.6.3-py36he774522_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\py-1.8.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pydensecrf-1.0rc3-py36h830ac7b_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pyparsing-2.4.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pyreadline-2.1-py36_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pyside2-5.9.0a1-py36h0555040_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pytest-5.0.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\python-3.6.8-h9f7ef89_7\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\python-dateutil-2.8.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pytz-2019.1-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pywavelets-1.0.3-py36h8c2d366_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\pyyaml-5.1.1-py36he774522_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\qt-5.9.7-vc14h73c81de_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\scikit-image-0.15.0-py36ha925a31_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\scikit-learn-0.19.1-py36hae9bb9f_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\scipy-1.1.0-py36h4f6bf74_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\seaborn-0.9.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\setuptools-41.0.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\shapely-1.6.4-py36h222a598_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\sip-4.19.8-py36h6538335_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\six-1.12.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\sortedcontainers-2.1.0-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\sqlalchemy-1.3.5-py36he774522_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\sqlite-3.29.0-he774522_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\statsmodels-0.10.0-py36h8c2d366_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tblib-1.4.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tensorboard-1.13.1-py36h33f27b4_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tensorflow-1.13.1-mkl_py36hd212fbe_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tensorflow-base-1.13.1-mkl_py36hcaf7020_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tensorflow-estimator-1.13.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\termcolor-1.1.0-py36_1\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tk-8.6.8-hfa6e2cd_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\toolz-0.10.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tornado-5.0.2-py36hfa6e2cd_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\tqdm-4.32.1-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\vc-14.1-h0510ff6_4\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\vs2015_runtime-14.15.26706-h3a45250_4\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\waitress-1.2.1-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\wcwidth-0.1.7-py36h3d5aa90_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\werkzeug-0.15.4-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\wheel-0.33.4-py36_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\wincertstore-0.2-py36h7fe50ca_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\xz-5.2.4-h2fa13f4_4\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\yaml-0.1.7-hc54c509_2\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\zict-1.0.0-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\zipp-0.5.1-py_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\zlib-1.2.11-h62dcd97_3\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs\zstd-1.3.7-h508b16e_0\info
Extract: repodata_record.json
Output folder: C:\Users\m1fsche\test8\pkgs
Extract: python-3.6.8-h9f7ef89_7.conda
Extract: _tflow_select-2.3.0-mkl.conda
Extract: blas-1.0-mkl.conda
Extract: ca-certificates-2019.5.15-0.conda
Extract: icc_rt-2019.0.0-h0cc432a_1.conda
Extract: intel-openmp-2019.4-245.conda
Extract: llvmdev-8.0.0-h1acfa63_0.conda
Extract: vs2015_runtime-14.15.26706-h3a45250_4.conda
Extract: libmklml-2019.0.3-0.conda
Extract: mkl-2018.0.3-1.conda
Extract: vc-14.1-h0510ff6_4.conda
Extract: clangdev-8.0.0-h1ad3211_2.tar.bz2
Extract: geos-3.7.1-h33f27b4_0.conda
Extract: icu-58.2-ha66f8fd_1.conda
Extract: jpeg-9b-hb83a4c4_2.conda
Extract: libiconv-1.15-h1df5818_7.conda
Extract: openssl-1.1.1c-he774522_1.conda
Extract: sqlite-3.29.0-he774522_0.tar.bz2
Extract: tk-8.6.8-hfa6e2cd_0.conda
Extract: xz-5.2.4-h2fa13f4_4.conda
Extract: yaml-0.1.7-hc54c509_2.conda
Extract: zlib-1.2.11-h62dcd97_3.conda
Extract: hdf5-1.10.4-h7ebc959_0.conda
Extract: libpng-1.6.37-h2a8f88b_0.conda
Extract: libprotobuf-3.8.0-h7bd577a_0.conda
Extract: libxml2-2.9.9-h464c3ec_0.conda
Extract: zstd-1.3.7-h508b16e_0.conda
Extract: astor-0.7.1-py36_0.conda
Extract: atomicwrites-1.3.0-py36_1.conda
Extract: attrs-19.1.0-py36_1.conda
Extract: blinker-1.4-py36_0.conda
Extract: certifi-2019.6.16-py36_0.conda
Extract: click-7.0-py36_0.conda
Extract: cloudpickle-1.2.1-py_0.tar.bz2
Extract: colorama-0.4.1-py36_0.conda
Extract: dask-core-0.17.4-py36_0.conda
Extract: decorator-4.4.0-py36_1.conda
Extract: freetype-2.9.1-ha9979f8_1.conda
Extract: gast-0.2.2-py36_0.conda
Extract: heapdict-1.0.0-py36_2.conda
Extract: itsdangerous-1.1.0-py36_0.conda
Extract: joblib-0.13.2-py36_0.conda
Extract: kiwisolver-1.1.0-py36ha925a31_0.conda
Extract: libtiff-4.0.10-hb898794_2.conda
Extract: libxslt-1.1.33-h579f668_0.conda
Extract: locket-0.2.0-py36hfed976d_1.conda
Extract: markupsafe-1.1.1-py36he774522_0.conda
Extract: marshmallow-3.0.0b8-py36_0.conda
Extract: more-itertools-7.0.0-py36_0.conda
Extract: msgpack-python-0.6.1-py36h74a9793_1.conda
Extract: numpy-base-1.15.4-py36h8128ebf_0.conda
Extract: olefile-0.46-py36_0.conda
Extract: pox-0.2.6-py_0.tar.bz2
Extract: psutil-5.6.3-py36he774522_0.conda
Extract: py-1.8.0-py36_0.conda
Extract: pydensecrf-1.0rc3-py36h830ac7b_1.tar.bz2
Extract: pyparsing-2.4.0-py_0.tar.bz2
Extract: pyreadline-2.1-py36_1.conda
Extract: pytz-2019.1-py_0.tar.bz2
Extract: pyyaml-5.1.1-py36he774522_0.conda
Extract: qt-5.9.7-vc14h73c81de_0.conda
Extract: sip-4.19.8-py36h6538335_0.conda
Extract: six-1.12.0-py36_0.conda
Extract: sortedcontainers-2.1.0-py36_0.conda
Extract: sqlalchemy-1.3.5-py36he774522_0.conda
Extract: tblib-1.4.0-py_0.tar.bz2
Extract: termcolor-1.1.0-py36_1.conda
Extract: toolz-0.10.0-py_0.tar.bz2
Extract: tornado-5.0.2-py36hfa6e2cd_0.conda
Extract: tqdm-4.32.1-py_0.tar.bz2
Extract: waitress-1.2.1-py36_0.conda
Extract: wcwidth-0.1.7-py36h3d5aa90_0.conda
Extract: werkzeug-0.15.4-py_0.tar.bz2
Extract: wincertstore-0.2-py36h7fe50ca_0.conda
Extract: zipp-0.5.1-py_0.tar.bz2
Extract: absl-py-0.7.1-py36_0.conda
Extract: cycler-0.10.0-py36h009560c_0.conda
Extract: cytoolz-0.10.0-py36he774522_0.tar.bz2
Extract: dill-0.2.9-py36_0.conda
Extract: importlib_metadata-0.17-py36_1.conda
Extract: mock-3.0.5-py36_0.conda
Extract: packaging-19.0-py36_0.conda
Extract: partd-1.0.0-py_0.tar.bz2
Extract: pillow-6.1.0-py36hdc69c19_0.tar.bz2
Extract: ppft-1.6.6.1-py36_0.tar.bz2
Extract: pyside2-5.9.0a1-py36h0555040_0.tar.bz2
Extract: python-dateutil-2.8.0-py36_0.conda
Extract: setuptools-41.0.1-py36_0.conda
Extract: zict-1.0.0-py_0.tar.bz2
Extract: cython-0.29.12-py36ha925a31_0.tar.bz2
Extract: distributed-1.21.8-py36_0.conda
Extract: grpcio-1.16.1-py36h351948d_1.conda
Extract: jinja2-2.10.1-py36_0.conda
Extract: markdown-3.1.1-py36_0.conda
Extract: multiprocess-0.70.8-py36hfa6e2cd_0.tar.bz2
Extract: networkx-2.3-py_0.tar.bz2
Extract: pluggy-0.12.0-py_0.tar.bz2
Extract: protobuf-3.8.0-py36h33f27b4_0.conda
Extract: wheel-0.33.4-py36_0.conda
Extract: flask-1.1.1-py_0.tar.bz2
Extract: pathos-0.2.4-py_0.tar.bz2
Extract: pip-19.1.1-py36_0.conda
Extract: pytest-5.0.1-py36_0.tar.bz2
Extract: bokeh-1.3.0-py36_0.tar.bz2
Extract: h5py-2.9.0-py36h5e291fa_0.conda
Extract: imageio-2.5.0-py36_0.conda
Extract: keras-applications-1.0.8-py_0.tar.bz2
Extract: matplotlib-3.1.0-py36hc8f65d3_0.conda
Extract: mkl_fft-1.0.6-py36hdbbee80_0.conda
Extract: mkl_random-1.0.1-py36h77b88f5_1.conda
Extract: numpy-1.15.4-py36ha559c80_0.conda
Extract: opencv-3.3.1-py36h20b85fd_1.conda
Extract: pandas-0.24.2-py36ha925a31_0.conda
Extract: pywavelets-1.0.3-py36h8c2d366_1.conda
Extract: scipy-1.1.0-py36h4f6bf74_1.conda
Extract: shapely-1.6.4-py36h222a598_0.conda
Extract: tensorboard-1.13.1-py36h33f27b4_0.conda
Extract: tensorflow-estimator-1.13.0-py_0.tar.bz2
Extract: dask-0.17.4-py36_0.conda
Extract: keras-preprocessing-1.1.0-py_1.tar.bz2
Extract: patsy-0.5.1-py36_0.conda
Extract: scikit-image-0.15.0-py36ha925a31_0.conda
Extract: scikit-learn-0.19.1-py36hae9bb9f_0.conda
Extract: imgaug-0.2.9-py_1.tar.bz2
Extract: keras-base-2.2.4-py36_0.conda
Extract: statsmodels-0.10.0-py36h8c2d366_0.conda
Extract: tensorflow-base-1.13.1-mkl_py36hcaf7020_0.conda
Extract: albumentations-0.3.0-py_0.tar.bz2
Extract: seaborn-0.9.0-py36_0.conda
Extract: tensorflow-1.13.1-mkl_py36hd212fbe_0.conda
Extract: keras-2.2.4-0.conda
Cannot open C:\Users\m1fsche\test8\_conda-script.py
Cannot open C:\Users\m1fsche\test8\_conda-script.py
Delete file: C:\Users\m1fsche\test8\pkgs\env.txt
Output folder: C:\Users\m1fsche\test8
Output folder: C:\Users\m1fsche\test8\conda-meta
Extract: history
Creating ZeissPyDefault menus...
Execute: "C:\Users\m1fsche\test8\pythonw.exe" -E -s "C:\Users\m1fsche\test8\Lib\_nsis.py" mkmenus 
Execute: "C:\Users\m1fsche\test8\pythonw.exe" -E -s "C:\Users\m1fsche\test8\Lib\_nsis.py" mkdirs
Running post install...
Execute: "C:\Users\m1fsche\test8\pythonw.exe" -E -s "C:\Users\m1fsche\test8\Lib\_nsis.py" post_install
Created uninstaller: C:\Users\m1fsche\test8\Uninstall-ZeissPyDefault.exe
Completed

```

The installer executed and finished successfully (see parts of creation log below -> Let me know if you need more details. I just did not want to paste that many lines of log output):

```
sys.version: 3.6.8 \Anaconda, Inc.\ (default, Feb 21  ... 
constructor version: 3.0.0   
conda interface type: conda   
 
fetching: python-dateutil-2.8.0-py36_0.conda   
fetching: setuptools-41.0.1-py36_0.conda   
fetching: zict-1.0.0-py_0.tar.bz2   
fetching: cython-0.29.12-py36ha925a31_0.tar.bz2   
fetching: distributed-1.21.8-py36_0.conda   
fetching: grpcio-1.16.1-py36h351948d_1.conda   
fetching: jinja2-2.10.1-py36_0.conda   
fetching: markdown-3.1.1-py36_0.conda   
fetching: multiprocess-0.70.8-py36hfa6e2cd_0.tar.bz2   
fetching: networkx-2.3-py_0.tar.bz2   
fetching: pluggy-0.12.0-py_0.tar.bz2   
fetching: protobuf-3.8.0-py36h33f27b4_0.conda   
fetching: wheel-0.33.4-py36_0.conda   
fetching: flask-1.1.1-py_0.tar.bz2   
fetching: pathos-0.2.4-py_0.tar.bz2   
fetching: pip-19.1.1-py36_0.conda   
fetching: pytest-5.0.1-py36_0.tar.bz2   
fetching: bokeh-1.3.0-py36_0.tar.bz2   
fetching: h5py-2.9.0-py36h5e291fa_0.conda   
fetching: imageio-2.5.0-py36_0.conda   
fetching: keras-applications-1.0.8-py_0.tar.bz2   
fetching: matplotlib-3.1.0-py36hc8f65d3_0.conda   
fetching: mkl_fft-1.0.6-py36hdbbee80_0.conda   
fetching: mkl_random-1.0.1-py36h77b88f5_1.conda   
fetching: numpy-1.15.4-py36ha559c80_0.conda   
fetching: opencv-3.3.1-py36h20b85fd_1.conda   
fetching: pandas-0.24.2-py36ha925a31_0.conda   
fetching: pywavelets-1.0.3-py36h8c2d366_1.conda   
fetching: scipy-1.1.0-py36h4f6bf74_1.conda   
fetching: shapely-1.6.4-py36h222a598_0.conda   
fetching: tensorboard-1.13.1-py36h33f27b4_0.conda   
fetching: tensorflow-estimator-1.13.0-py_0.tar.bz2   
fetching: dask-0.17.4-py36_0.conda   
fetching: keras-preprocessing-1.1.0-py_1.tar.bz2   
fetching: patsy-0.5.1-py36_0.conda   
fetching: scikit-image-0.15.0-py36ha925a31_0.conda   
fetching: scikit-learn-0.19.1-py36hae9bb9f_0.conda   
fetching: imgaug-0.2.9-py_1.tar.bz2   
fetching: keras-base-2.2.4-py36_0.conda   
fetching: statsmodels-0.10.0-py36h8c2d366_0.conda   
fetching: tensorflow-base-1.13.1-mkl_py36hcaf7020_0.conda   
fetching: albumentations-0.3.0-py_0.tar.bz2   
fetching: seaborn-0.9.0-py36_0.conda   
fetching: tensorflow-1.13.1-mkl_py36hd212fbe_0.conda   
fetching: keras-2.2.4-0.conda   
Checking for duplicate files ...   
Checking for 'c:\miniconda\envs\lala\NSIS\makensis.exe'   
NSIS version: v3.01   
Reading: c:\miniconda\envs\lala\lib\site-packages\constructor\nsis\main.nsi.tmpl   
Created C:\Users\VSSADM~1\AppData\Local\Temp\tmph80cjn2h\main.nsi file   
Calling: ['c:\\miniconda\\envs\\lala\\NSIS\\makensis.exe', '/V4', 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\tmph80cjn2h\\main.nsi']   
makensis stdout:   
MakeNSIS v3.01 - Copyright 1999-2016 Contributors

...


Install: 7 pages (448 bytes), 1 section (2072 bytes), 2042 instructions (57176 bytes), 3532 strings (35562 bytes), 1 language table (322 bytes).   
--  --
       
   Uninstall: 4 pages (320 bytes), 1 section (2072 bytes), 434 instructions (12152 bytes), 778 strings (6652 bytes), 1 language table (290 bytes).   
       
   Datablock optimizer saved 58064 bytes (~0.0%).   
       
       
       
   Using zlib compression.   
       
       
       
   EXE header size:               58880 / 38912 bytes   
       
   Install code:                  96072 / 96068 bytes   
       
   Install data:              849203065 / 849271500 bytes   
       
   Uninstall code+data:          138900 / 138892 bytes   
       
   CRC (0xEDC097BE):                  4 / 4 bytes   
       
       
       
   Total size:                849496921 / 849545376 bytes (99.9%)   
       
       
   makensis stderr:   
       
   Successfully created 'd:\a\1\s\Common\InstallEnvironment\installer\lalaCPU20190724.exe'.
```

What might be of interest is that there is no __conda-script.py_ mentioned in the creation log. Other files that are listed in the installation log are also listed in the creation log.

I tested with and without elevated user rights.

I appreciate any hints that might lead to a solution!

**Comment by nehaljwani** [comment]
@FelixS90 I have pushed a few more commits to v3 branch. Could you please try again? If you still face an issue, could you please share your construct.yaml?

**Comment by FelixS90** [comment]
@nehaljwani Thanks a lot for your quick reply!

> Could you please try again?

I just pulled your latest changes, but **the error unfortunately remains the same**.

> If you still face an issue, could you please share your construct.yaml?

```
name: lala

version: 20190724

installer_filename: lalaCPU20190724.exe

installer_type: exe

channels:
  - http://repo.continuum.io/pkgs/main
  - http://repo.continuum.io/pkgs/free
  - http://repo.continuum.io/pkgs/msys2
  - https://conda.anaconda.org/conda-forge

specs:
  - python ==3.6.*
  - albumentations
  - atomicwrites
  - blinker
  - cython
  - cloudpickle
  - dask ==0.17.4
  - distributed ==1.21.8
  - flask
  - joblib
  - keras
  - marshmallow
  - matplotlib
  - numpy 
  - pandas
  - pathos
  - pillow
  - pox
  - psutil
  - pydensecrf
  - pyside2
  - pytest
  - pyyaml
  - scikit-image 
  - scikit-learn ==0.19.1
  - scipy
  - seaborn
  - sqlalchemy
  - tensorflow ==1.13.1
  - tornado ==5.0.2
  - tqdm
  - waitress
  - werkzeug

exclude:
  - pyqt

install_in_dependency_order: True

initialize_by_default: False

register_python_default: False

keep_pkgs: False

license_file: ..\License_blank.txt

welcome_image: ..\icons\lala_welcome.png

header_image: ..\icons\lala_header.png

icon_image: ..\icons\lala_RGB.png

company: lala
```
The output from my environment's `conda list` is
```
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py36_0
bzip2                     1.0.8                he774522_0
ca-certificates           2019.5.15                     0
certifi                   2019.6.16                py36_0
cffi                      1.12.3           py36h7a1dbc1_0
chardet                   3.0.4                    py36_1
conda                     4.7.10                   py36_0
conda-package-handling    1.3.11                   py36_0
constructor               3.0.0                    pypi_0    pypi
cryptography              2.7              py36h7a1dbc1_0
git                       2.20.1               h6bb4b03_0
idna                      2.8                      py36_0
libarchive                3.3.3                h0643e63_5
libiconv                  1.15                 h1df5818_7
libxml2                   2.9.9                h464c3ec_0
lz4-c                     1.8.1.2              h2fa13f4_0
lzo                       2.10                 h6df0209_2
menuinst                  1.4.16           py36he774522_0
nsis                      3.01                          8
openssl                   1.1.1c               he774522_1
pillow                    6.1.0                    pypi_0    pypi
pip                       19.1.1                   py36_0
pycosat                   0.6.3            py36hfa6e2cd_0
pycparser                 2.19                     py36_0
pyopenssl                 19.0.0                   py36_0
pysocks                   1.7.0                    py36_0
python                    3.6.8                h9f7ef89_7
python-libarchive-c       2.8                     py36_11
pywin32                   223              py36hfa6e2cd_1
pyyaml                    5.1.1            py36he774522_0
requests                  2.22.0                   py36_0
ruamel_yaml               0.15.46          py36hfa6e2cd_0
setuptools                41.0.1                   py36_0
six                       1.12.0                   py36_0
sqlite                    3.29.0               he774522_0
tqdm                      4.32.1                     py_0
urllib3                   1.24.2                   py36_0
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.15.26706          h3a45250_4
wheel                     0.33.4                   py36_0
win_inet_pton             1.1.0                    py36_0
wincertstore              0.2              py36h7fe50ca_0
xz                        5.2.4                h2fa13f4_4
yaml                      0.1.7                hc54c509_2
zlib                      1.2.11               h62dcd97_3
zstd                      1.3.7                h508b16e_0
```
I created it using
`conda create -n "lala" -y -m --channel=http://repo.continuum.io/pkgs/main/ --channel=http://repo.continuum.io/pkgs/free/ --channel=http://repo.continuum.io/pkgs/msys2 --override-channels python=3.6 conda pyyaml git pip nsis
`
After activating the environment, I do a
`pip install git+https://github.com/conda/constructor.git@v3`
and finally
```
constructor --clean --conda-exe %CONDA_EXE% ".\path_to_construct"
constructor --version --conda-exe %CONDA_EXE% ".\path_to_construct"
constructor --test --conda-exe %CONDA_EXE% ".\path_to_construct"
constructor --platform=win-64 -v --conda-exe %CONDA_EXE% --output-dir %OUTPUT_DIR% ".\path_to_construct"
```
where 
- `CONDA_EXE` is the path to the [conda executable](https://repo.anaconda.com/pkgs/misc/conda-execs/conda-latest-win-64.exe) downloaded to disk
- OUTPUT_DIR is the path to the directory holding the installer.

Does that help?

**Comment by ankostis** [comment]
@FelixS90 i _think_ i had a similar problem (incomlete installations).

Everything was fixed when i reinstalled a clean miniconda-base.
I _think_ that Windows don't like `conda update conda` to upgrade to the latest conda, at least in this case.

**Comment by msarahan** [comment]
added in #263.  Please open new, more specific issues for any problems you encounter with the new code.
